pkg/kubelet/kubeletconfig/util/test/test.go

```diff
@@ -41,15 +43,17 @@ func ExpectError(t *testing.T, err error, substr string) {
 // The return value indicates whether we should skip the rest of the test case due to the error result.
 func SkipRest(t *testing.T, desc string, err error, contains string) bool {
 	if err != nil {
+		// We got an error ...
```

Comment:
bugfix + this is probably simpler:
```go
// return bools are negated because we skip when asserts fail, not when they succeed
if contains == "" {
  return !assert.NoError // ...
}
// expect error containing str
return !assert.Error(t, err, "case %q, expected err containing %q, but got nil error", desc, contains ) ||
       !assert.Contains // ...
```

Reply:
Another option:
```go
if contains == "" {
  assert.NoError // ...
} else {
  assert.Error(t, err, "case %q, expected err containing %q, but got nil error", desc, contains)
  assert.Contains // ...
}
return t.Failed()
```

Reply:
Keep in mind that the assert lib already prints nice error messages.
We're also using `t.Run` which will group failures with the named test.
There's no need to format custom messages unless you have special context.
ex:
```
--- FAIL: TestContains (0.00s)
    --- FAIL: TestContains/#00 (0.00s)
        lib_test.go:26: 
            	Error Trace:	lib_test.go:26
            	Error:      	An error is expected but got nil.
            	Test:       	TestContains/#00
    --- FAIL: TestContains/new_world (0.00s)
        lib_test.go:27: 
            	Error Trace:	lib_test.go:27
            	Error:      	"new world" does not contain "hello"
            	Test:       	TestContains/new_world
            	Messages:   	new world
```

Reply:
Thanks! I simplified it.